### PR TITLE
Add todo list example

### PR DESCRIPTION
**Pull request by GShotwell**
This is a basic todo list app somewhat inspired by the [TodoMVC](https://todomvc.com/) challenge.

### CODE REVIEW

File: shiny/examples/todo_list/app.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+    @reactive.Effect
+    @reactive.event(input.add)
+    def add():
+        counter = task_counter.get() + 1
+        task_counter.set(counter)
+        id = "task_" + str(counter)
+        ui.insert_ui(
+            selector="#tasks",
+            where="afterBegin",
```
Comment (jcheng5) [inline shiny/examples/todo_list/app.py:54]: This is the behavior I personally would expect from a todo app
```suggestion
            where="beforeEnd",
```

File: shiny/examples/todo_list/app.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            border-top: 1px solid #eeeeee;
+            margin-top: -1px;
+            padding: 20px 10px;
+            align-items: center;
+        }
+        .row p {
+        margin-bottom: 0px;
+        }
+        """
+    ),
```
Comment (jcheng5) [inline shiny/examples/todo_list/app.py:21]: With the changes below, these styles don't seem to be needed
```suggestion
```

File: shiny/examples/todo_list/app.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        id = "task_" + str(counter)
+        ui.insert_ui(
+            selector="#tasks",
+            where="afterBegin",
+            ui=task_ui(id),
+        )
+
+        # Since we want the action button inside the module to update
+        # the finished_tasks global counter, we need to pass the reactive values list
+        # down to the module function.
```
Comment (jcheng5) [inline shiny/examples/todo_list/app.py:60]: I think there's a bit too much coupling between the `task_server` internals and the calling code. I don't think it's a ship-stopper for this example, but if you're interested we could do a refactoring in 15 minutes on zoom.

File: shiny/examples/todo_list/app.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        ui.update_text("todo_input_text", value="")
+
+
+# Modules to define the rows
+
+
+@module.ui
+def task_ui():
+    out = ui.output_ui("button_row")
+    return out
```
Comment (jcheng5) [inline shiny/examples/todo_list/app.py:73]:
```suggestion
    return ui.output_ui("button_row")
```

File: shiny/examples/todo_list/app.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+                ui.p(text, style=text_style),
+            ),
+        )
+        return out
+
+    @reactive.Effect
+    @reactive.event(input.finish)
+    def finish_task():
+        task_list.set(task_list.get() + 1)
+        finished.set(False)
```
Comment (jcheng5) [inline shiny/examples/todo_list/app.py:111]:
```suggestion
        finished.set(True)
```

**Review by jcheng5 — CHANGES_REQUESTED** [review]

File: shiny/examples/todo_list/app.py (hunk 3 above)
Comment (GShotwell) [reply shiny/examples/todo_list/app.py:46]: Sure, this was the bit I liked the least so I'd love to learn a better pattern.

**Pull request merged**